### PR TITLE
BR Nightly setup and updated release workflow

### PR DESCRIPTION
**Pull request by nbhasin2**
Renamed Release workflow and added a Nightly workflow 
Added three new workflows
- Deploy_Nightly
- Deploy_Beta
- Deploy_Prod_Beta

### CODE REVIEW

File: Shared/AppInfo.swift [hunk 1]
```diff
@@ -78,7 +78,12 @@ open class AppInfo {
 
     // Return the MozWhatsNewTopic key from the Info.plist
     public static var whatsNewTopic: String? {
-        return Bundle.main.object(forInfoDictionaryKey: "MozWhatsNewTopic") as? String
+        let appVersionSplit = AppInfo.appVersion.components(separatedBy: ".")
+        let majorAppVersion = appVersionSplit[0]
+        var dotVersion = ""
+        if appVersionSplit.count > 1, appVersionSplit[0] != "0" { dotVersion = appVersionSplit[1] }
+        let topic = "whats-new-ios-\(majorAppVersion)\(dotVersion)"
```
Comment (nbhasin2) [inline Shared/AppInfo.swift:85]: remove `dotVersion` from topic if we don't want to use it for whats new section

File: Client/Info.plist [hunk 2]
```diff
@@ -102,6 +102,8 @@
 	<string>$(POCKET_API_KEY)</string>
 	<key>SentryDSN</key>
 	<string>$(SENTRY_DSN)</string>
+	<key>SentryNightlyDSN</key>
+	<string>$(SENTRY_NIGHTLY_DSN)</string>
```
Comment (nbhasin2) [inline Client/Info.plist:106]: Added a nightly specific DSN for sentry

**Review by isabelrios — APPROVED** [review]

File: bitrise.yml [hunk 3]
```diff
@@ -755,10 +1136,22 @@ app:
...
+    BITRISE_BETA_VERSION: '95'
 trigger_map:
 - push_branch: main
   workflow: RunUnitTests
 - push_branch: v39.0
-  workflow: xcode13-release-and-beta-nocache
+  workflow: Deploy_Prod_Beta
 - pull_request_target_branch: main
   workflow: RunUnitTests
+meta:
```
Comment (isabelrios) [inline bitrise.yml:1155]: Not sure this is needed as this is setting a general stack while we have each stack defined per workflow which is used insteao of this one.

**Comment by nbhasin2** [comment]
Merging it in as last commit has very minor changes related to bumping version and cleanup

https://github.com/mozilla-mobile/firefox-ios/pull/9413/commits/fd7d9888550a47e7ec47e8189386ea9f9ed1e030

**Pull request merged**